### PR TITLE
support for SecOC cars

### PR DESCRIPTION
**Pull request by jyoung8607**
Separating the SecOC support code from the RAV4 Prime port.

- [x] commaai/opendbc#1311

### CODE REVIEW

**Comment by jyoung8607** [comment]
@adeebshihadeh Are you happy with this way of handling keys?

File: selfdrive/car/card.py [hunk 1]
```diff
@@ -127,6 +127,17 @@ def __init__(self, CI=None, RI=None) -> None:
       safety_config.safetyModel = structs.CarParams.SafetyModel.noOutput
       self.CP.safetyConfigs = [safety_config]
 
+    secoc_key = self.params.get("SecOCKey", encoding='utf8')
```
Comment (adeebshihadeh) [inline selfdrive/car/card.py:130]: let's also gate this on a development build for now, like we do with alpha long support

**Review by adeebshihadeh — APPROVED** [review]
LGTM, also left a minor comment on the opendbc PR.

Not sold that we should show a special alert for this, but we can start with it.

**Comment by jyoung8607** [comment]
> Not sold that we should show a special alert for this, but we can start with it.

IMO anything that helps make dashcam reasons clear is a good thing. Otherwise new users will show up in Discord over and over with the same question, plus they'll spend time going down a fingerprinting rabbit hole. We also need to alert over top of LKA Fault when the SecOC key is missing.

With the new release branch gating, I'm actually on the fence whether we should have another alert (or move to a configurable text alert) for the prevented SecOC startup, but this will do for now.

**Review by adeebshihadeh — APPROVED** [review]

**Pull request merged**